script.js

```diff
@@ -1,29 +1,64 @@
 function setup() {
   const allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
+  searchEpisodeCards();
 }
 
 function makePageForEpisodes(allEpisodes) {
   const rootElem = document.getElementById("root");
-  const episodeList =  allEpisodes.map(createEpisodeCard)
-  rootElem.append(...episodeList)
+  const episodeList = allEpisodes.map(createEpisodeCard);
+  rootElem.append(...episodeList);
 }
 
-function formatSeason(season){
-  return season  < 10 ? `S0${season}` : `S${season}`
+function formatSeason(season) {
+  return season < 10 ? `S0${season}` : `S${season}`;
 }
 
-function formatEpisode(episode){
-  return episode  < 10 ? `E0${episode}` : `E${episode}`
+function formatEpisode(episode) {
+  return episode < 10 ? `E0${episode}` : `E${episode}`;
 }
 
-function createEpisodeCard(episode){
-  const episodeCard = document.querySelector('#episode-card-template').content.cloneNode(true)
-  episodeCard.querySelector('h1').textContent = episode.name
-  episodeCard.querySelector('[data-season-episode]').textContent = `- ${formatSeason(episode.season)}${formatEpisode(episode.number)}`
-  episodeCard.querySelector('img').setAttribute('src', episode.image.medium)
-  episodeCard.querySelector('p').textContent = episode.summary.replace(/<p>|<\/p>/g, '')
-  return episodeCard  
+function createEpisodeCard(episode) {
+  const episodeCard = document
+    .querySelector("#episode-card-template")
+    .content.cloneNode(true);
+  episodeCard.querySelector("h1").textContent = episode.name;
+  episodeCard.querySelector(
+    "[data-season-episode]"
+  ).textContent = `- ${formatSeason(episode.season)}${formatEpisode(
+    episode.number
+  )}`;
+  episodeCard.querySelector("img").setAttribute("src", episode.image.medium);
+  episodeCard.querySelector("p").textContent = episode.summary.replace(
+    /<p>|<\/p>/g,
+    ""
+  );
+  return episodeCard;
 }
 
+function searchEpisodeCards(allEpisodes) {
```

Comment:
Could you please elaborate why you have ```allEpisodes``` as a parameter here? why are you not using it in your code and where the ```searchEpisodeCards``` is called?

Reply:
Including allEpisodes as a parameter ensures that the function has access to the list of cards so after the search rerender the list. I needed to pass it due to the way you create cards by template .

Reply:
OK I see! But do you think the ```allEpisodes``` param has any effect here since you are not using it inside the function? also you are not passing it when you called the ```searchEpisodeCards``` function in the ```setup``` function?

Reply:
Right, you are absolutely correct to notice that the allEpisodes parameter in the searchEpisodeCards is unnecessary and definitely needs to be removed.